core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -2754,6 +2754,8 @@ class KafkaApis(val requestChannel: RequestChannel,
           authHelper.authorize(originalRequest.context, ALTER_CONFIGS, CLUSTER, CLUSTER_NAME)
         case ConfigResource.Type.TOPIC =>
           authHelper.authorize(originalRequest.context, ALTER_CONFIGS, TOPIC, resource.name)
+        case ConfigResource.Type.CLIENT_METRICS =>
+          authHelper.authorize(originalRequest.context, ALTER_CONFIGS, CLIENT_METRICS, resource.name)
```

Comment:
See earlier comment. There are no ACLs for `CLIENT_METRICS` resource type and checking is performed against the `CLUSTER` resource.

Reply:
Completely missed it, thanks for that. I have removed declaring new Resource Type for client_metrics and corresponding checks. Have changed test cases to authorize on CLUSTER resource for CLIENT_METRICS config source.